apis/v1alpha2/httproute_types.go

```diff
@@ -282,6 +283,23 @@ const (
 	HeaderMatchImplementationSpecific HeaderMatchType = "ImplementationSpecific"
 )
 
+// HTTPHeaderName is the name of an HTTP header.
+//
+// Valid values include:
+//
+// * Authorization
+// * Set-Cookie
+//
+// Invalid values include:
+//
+// * :method
```

Comment:
quotes, say why it's invalid (":", "/" are not a valid character for header names)

Reply:
We should explicitly call out that psuedo headers are not allowed.

Reply:
Updated, thanks!